src/seedsigner/views/seed_views.py

```diff
@@ -2294,3 +2298,419 @@ def run(self):
+            combined_entropy = bip39.mnemonic_to_bytes(first_shard.mnemonic_str)
+            
+            for i in range(1, len(self.controller.rebuild_seedxor_shards)):
+                next_shard = self.controller.rebuild_seedxor_shards[i]
+                next_entropy = bip39.mnemonic_to_bytes(next_shard.mnemonic_str)
+                combined_entropy = xor_bytes(combined_entropy, next_entropy)
+            
+            combined_mnemonic = bip39.mnemonic_from_bytes(combined_entropy).split()
+            self.controller.rebuild_seedxor_combined_seed = Seed(mnemonic=combined_mnemonic)
+            self.controller.storage.set_pending_seed(self.controller.rebuild_seedxor_combined_seed)
+            self.seed = self.controller.storage.get_pending_seed()
+            
+            self.fingerprint = self.seed.get_fingerprint(
+                network=self.settings.get_value(SettingsConstants.SETTING__NETWORK)
+            )
+        except Exception as e:
+            self.error = str(e)
+    
+    def run(self):
+        if self.error == "no_shards":
+            return Destination(RebuildSeedXORManageView)
+        elif self.error:
+            from seedsigner.views.view import ErrorView
+            return Destination(ErrorView, view_args=dict(
+                title="XOR Error",
+                status_headline="Error Combining Seeds",
+                text=f"Error: {self.error}",
+                button_text="OK",
+                next_destination=Destination(RebuildSeedXORManageView)
+            ))
+        
+        shard_count = len(self.controller.rebuild_seedxor_shards)
+        button_data = [ButtonOption("Continue")]
+        
+        from seedsigner.gui.screens.screen import LargeIconStatusScreen
+        selected_menu_num = self.run_screen(
+            LargeIconStatusScreen,
+            title=_(f"Finalize SeedXOR"),
+            status_headline=_(f"Combined {shard_count} shards"),
+            text=_(f"Fingerprint: {self.fingerprint}\n\nCombined seed calculated successfully."),
+            button_data=button_data,
+        )
+        
+        if selected_menu_num == RET_CODE__BACK_BUTTON:
+            return Destination(RebuildSeedXORManageView)
+        
+        return Destination(RebuildSeedXORFinalizeOptionsView)
+
+
+class RebuildSeedXORFinalizeOptionsView(View):
+    """Second step of finalization - choose what to do with individual shards"""
+    
+    DISCARD_SHARDS = ButtonOption("Discard Shards", button_label_color="red")
+    KEEP_SHARDS = ButtonOption("Keep Shards")
+    PASSPHRASE = ButtonOption("BIP-39 Passphrase")
+    
+    def __init__(self):
+        super().__init__()
+        self.error = False
+        
+        if not self.controller.rebuild_seedxor_combined_seed:
+            self.error = True
+        
+        self.seed = self.controller.storage.get_pending_seed()
+        
+    def run(self):
+        if self.error:
+            return Destination(RebuildSeedXORManageView)
```

Comment:
This workaround can be avoided by using View's `set_redirect()` in the `__init__()`.